### PR TITLE
feat(NODE-4509): send 1 with hello commands

### PR DESCRIPTION
**Pull request by durran**
### Description

Whenever the driver itself issues a `hello` or legacy hello, we now set the command value to `1` instead of `true`.

#### What is changing?
- Changes the commands to send `1` instead of `true`.
- Changes tests that also send the commands to the same.

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

NODE-4509

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/cmap/connect.ts [hunk 1]
```diff
@@ -231,7 +231,7 @@ export function prepareHandshakeDocument(
   const { serverApi } = authContext.connection;
 
   const handshakeDoc: HandshakeDocument = {
-    [serverApi?.version ? 'hello' : LEGACY_HELLO_COMMAND]: true,
+    [serverApi?.version ? 'hello' : LEGACY_HELLO_COMMAND]: 1,
```
Comment (nbbeeken) [inline src/cmap/connect.ts:234]: Is there a way we can get regression coverage here? It looks like we only have a test that asserts the monitor will use `hello: 1`.

File: test/integration/mongodb-handshake/promise_handshake.test.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ describe('Handshake', function () {
         // Execute legacy hello command
         client
           .db(configuration.db)
-          .command({ [LEGACY_HELLO_COMMAND]: true })
+          .command({ [LEGACY_HELLO_COMMAND]: 1 })
```
Comment (nbbeeken) [inline test/integration/mongodb-handshake/promise_handshake.test.js:31]: This test file doesn't add much value to our suite, + the strange constraints like poolsize, using legacy format strings, only on single, promise usage combined with mocha done style. Can we just delete this? 😅

File: src/cmap/connect.ts (hunk 1 above)
Comment (durran) [reply src/cmap/connect.ts:234]: Added 2 unit tests for this now.

File: test/integration/mongodb-handshake/promise_handshake.test.js (hunk 2 above)
Comment (durran) [reply test/integration/mongodb-handshake/promise_handshake.test.js]: Removed.

**Review by nbbeeken — APPROVED** [review]

**Review by baileympearson — APPROVED** [review]

**Pull request merged**